python-avd/pyavd/_eos_designs/structured_config/flows/__init__.py

```diff
@@ -124,70 +110,95 @@ def _enable_sflow(self) -> bool:
 
-        return tracker
+        This relies on flow-tracking being rendered after all other eos_designs modules (except structured config).
+        """
+        if self.shared_utils.flow_tracking_type == "hardware":
+            self._set_hardware_flow_tracking()
+        elif self.shared_utils.flow_tracking_type == "sampled":
+            self._set_sampled_flow_tracking()
 
-    @cached_property
-    def flow_tracking(self) -> dict | None:
-        """Return structured config for flow_tracking."""
-        configured_trackers = self._get_enabled_flow_trackers()
-        if not configured_trackers:
-            return None
-
-        flow_tracking = {}
-
-        tracker_type = self.shared_utils.flow_tracking_type
-        global_settings = self.inputs.flow_tracking_settings.hardware if tracker_type == "hardware" else self.inputs.flow_tracking_settings.sampled
-        flow_tracking[tracker_type] = global_settings._as_dict()
-        if tracker_type == "sampled":
-            # asdict does not contain default values so we need to insert the default sample.
-            # TODO: consider if asdict should include defaults.
-            flow_tracking[tracker_type]["sample"] = global_settings.sample
-
-        filtered_trackers = []
-        for tracker_name in natural_sort(configured_trackers):
-            """
-            We allow overriding the default flow tracker name, so if user has configured a tracker
-            with the default tracker name, then we just use that, if not, we create a default config
-            """
-            default_tracker = next(iter(EosDesigns.FlowTrackingSettings().trackers))
-            if tracker_name not in self.inputs.flow_tracking_settings.trackers:
-                if tracker_name == default_tracker.name:
-                    tracker = default_tracker
-                else:
-                    msg = f"{tracker_name} is being used for one of the interfaces, but is not configured in flow_tracking_settings"
-                    raise AristaAvdInvalidInputsError(msg)
-            else:
-                tracker = self.inputs.flow_tracking_settings.trackers[tracker_name]
+    def _set_hardware_flow_tracking(self) -> None:
+        """Set the structured configuration for hardware flow tracking if any interface is configured."""
+        all_interfaces = chain(
+            self.structured_config.ethernet_interfaces, self.structured_config.port_channel_interfaces, self.structured_config.dps_interfaces
+        )
 
-            filtered_trackers.append(self.resolve_flow_tracker_by_type(tracker))
+        trackers: set[str] = {interface.flow_tracker.hardware for interface in all_interfaces if interface.flow_tracker.hardware}
+        if not trackers:
+            return
+
+        self.structured_config.flow_tracking.hardware = self.inputs.flow_tracking_settings.hardware._cast_as(EosCliConfigGen.FlowTracking.Hardware)
```

Comment:
Can we try to change the schema to use a $ref when we also use cast_as? Otherwise a drift in schemas might lead to bad issues.